src/test/java/com/google/sps/MatchServletTest.java

```diff
@@ -35,8 +36,17 @@ public void doGet_returnMatches() throws IOException, ServletException {
         testRepository = repository;
         Gson gson = new Gson();
 
-        MatchServlet matchServlet = new MatchServlet();
-        matchServlet.init();
+        //mock ServletContext
+        final ServletContext servletContext = Mockito.mock(ServletContext.class);
+
+        //override getServletContext
+        MatchServlet matchServlet = new MatchServlet() {
+            public ServletContext getServletContext() {
+                return servletContext;
+            }
+        };
+        when(servletContext.getAttribute("matchRepository")).thenReturn(repository);
+        when(servletContext.getAttribute("currentUser")).thenReturn(testUser);
```

Comment:
Similar to above, you should be able to factor out the common portions (making the servletContext, setting up expectations) into an init method annotaded with a @BeforeEach

Reply:
Done!